src/librbd/cache/rwl/LogOperation.h

```diff
@@ -166,6 +172,35 @@ class WriteLogOperationSet {
                                   const WriteLogOperationSet &s);
 };
 
+class DiscardLogOperation : public GenericWriteLogOperation {
+public:
+  using GenericWriteLogOperation::m_lock;
+  using GenericWriteLogOperation::sync_point;
+  using GenericWriteLogOperation::on_write_append;
+  using GenericWriteLogOperation::on_write_persist;
+  std::shared_ptr<DiscardLogEntry> log_entry;
+  DiscardLogOperation(std::shared_ptr<SyncPoint> sync_point,
+                      const uint64_t image_offset_bytes,
+                      const uint64_t write_bytes,
+                      const utime_t dispatch_time,
+                      PerfCounters *perfcounter,
+                      CephContext *cct);
+  ~DiscardLogOperation() override;
+  DiscardLogOperation(const DiscardLogOperation&) = delete;
+  DiscardLogOperation &operator=(const DiscardLogOperation&) = delete;
+  const std::shared_ptr<GenericLogEntry> get_log_entry() override {
+    return log_entry;
+  }
+  bool reserved_allocated() override {
```

Comment:
Nit: (fix const-ness of virtual method)
```suggestion
  bool reserved_allocated() const override {
```

Reply:
Done